### PR TITLE
fix: pipeline error

### PR DESCRIPTION
**Pull request by kevinccbsg**
### Type:
* [ ] cicd: helm, docker & cicd adjustments.
* [ ] feature: new capabilities.
* [X] fix: bug, hotfix, etc.
* [ ] refactor: enhacements.
* [ ] style: changes in styles.
* [X] other: docs, tests.

### What's the focus of this PR:
There was an error in the docker-compose script as we were using the alpine version which it doesn't have bash installed so the docker-compose always fails.

We also change the Sonarqube cliProjectKey as it was using the pipeline name and this name can contains /. This name should be the one that we provide as a variable with yeoman.

Also, we added some documentation on the Sonarqube and Build image tasks as it was changed to use service connection instead of the hardcoded registry but it is not clear if you are downloading it for the first time what should we do to run the pipeline.

### CODE REVIEW

File: generators/app/templates/.helm/values.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 replicaCount: 1
 
 image:
-  repository: ilwebshops.azurecr.io/<%= name %>
+  repository: ilwebshops.azurecr.io/infinitaslearning/<%= name %>
```
Comment (danielherrerohernando) [inline generators/app/templates/.helm/values.yaml:8]: **ilwebshops.azurecr.io** is the name of the container registry that the **e-commerce team** uses, but this generator is supposed to be used by anyone within GS-Infinitas. 
I think we should add the name of the container registry as a parameter to the generator so you can provide it in the same way you provide a name.
`repository: <%= projectcontainerregistry %>/<%= name %>`

Comment (danielherrerohernando) [reply generators/app/templates/.helm/values.yaml:8]: `{
      type    : 'input',
      name    : 'projectcontainerregistry',
      message : 'Your project's container registry name (e.g. discoregistryinfinitas.azurecr.io)',
      default : ''
    }`

Comment (kevinccbsg) [reply generators/app/templates/.helm/values.yaml:8]: I think there are more places to add that variable. I think it is a good feature.

Although this is PR for a fix for the generator, I'll add that feature no problem 👍.

Comment (danielherrerohernando) [reply generators/app/templates/.helm/values.yaml:8]: Yes , that value is used in the docker-compose yml and in this helm file

Comment (kevinccbsg) [reply generators/app/templates/.helm/values.yaml]: I've just added it 😄

**Comment by kevinccbsg** [comment]
I think this package needs a pipeline for updating it to Infinitas artifact. I'll register some issues.

**Comment by kevinccbsg** [comment]
@danielherrerohernando I applied the change. Is it okay to merge this? 😄

**Comment by danielherrerohernando** [comment]
> @danielherrerohernando I applied the change. Is it okay to merge this? 😄

sure 🦑

**Pull request merged**